src/Xunit.StaFact/Sdk/UITestInvoker.cs

```diff
@@ -44,7 +44,7 @@ public new async Task<decimal> RunAsync()
                     if (!this.CancellationTokenSource.IsCancellationRequested)
                     {
                         await this.threadRental.SynchronizationContext;
-                        await this.BeforeTestMethodInvokedAsync();
+                        await this.Aggregator.RunAsync(this.BeforeTestMethodInvokedAsync);
```

Comment:
❔ Do we want AfterTestMethodInvokedAsync to be invoked or skipped if this fails? I lean towards skipping it if the aggregator has errors after the `Before` invocation.

Reply:
Perhaps the solution here is to wrap the entire block from `BeforeTestMethodInvokedAsync` to `AfterTestMethodInvokedAsync` in a new `Aggregator.RunAsync`, smiilar to the way `InitializeAsync` is handled.